airbyte-integrations/connectors/destination-google-sheets/destination_google_sheets/spreadsheet.py

```diff
@@ -0,0 +1,96 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+from typing import List, Mapping
+
+from pygsheets import Spreadsheet, Worksheet
+from pygsheets.client import Client as pygsheets_client
+from pygsheets.exceptions import WorksheetNotFound
+
+
+class GoogleSheets:
+    def __init__(self, client: pygsheets_client, spreadsheet_id: str):
+        self.client = client
+        self.spreadsheet_id = spreadsheet_id
+
+    @property
+    def spreadsheet(self) -> Spreadsheet:
+        """
+        Returns pygsheets.Spreadsheet with opened target spreadsheet by key.
+        """
+        return self.client.open_by_key(self.spreadsheet_id)
+
+    def open_worksheet(self, stream_name: str) -> Worksheet:
+        """
+        Opens the connection to target worksheet, if exists. Otherwise, creates one.
+        """
+        try:
+            stream = self.spreadsheet.worksheet_by_title(stream_name)
+        except WorksheetNotFound:
+            stream = self.spreadsheet.add_worksheet(stream_name)
+        finally:
+            return stream
```

Comment:
same here with the finally, we don't actually want to run that return statement if we throw a different exception so can take outside of the finally.

Reply:
This is different, the try/except is used here to perform the creation of the worksheet if it's not there. It's done by handling the `WorksheetNotFound` exception, instead of making more api calls to check it exists, we explicitly handle this case and return the `stream` in either way . But the `finally` block wasn't necessary indeed.